kalite/control_panel/api_resources.py

```diff
@@ -245,6 +245,16 @@ def alter_list_data_to_serialize(self, request, to_be_serialized):
             attempt_logs = AttemptLog.objects.filter(user=user, exercise_id=bundle.data["exercise_id"], context_type__in=["playlist", "exercise"])
             bundle.data["timestamp_first"] = attempt_logs.count() and attempt_logs.aggregate(Min('timestamp'))['timestamp__min'] or None
             bundle.data["timestamp_last"] = attempt_logs.count() and attempt_logs.aggregate(Max('timestamp'))['timestamp__max'] or None
+            bundle.data["unit"] = 0
+
+            if bundle.data["timestamp_first"]:
+                for i in xrange(101,104):
```

Comment:
Note also that in a few cases they switched into later units temporarily and then back, which means there could be gift cards for future units already.

It's like a jigsaw puzzle!!! :D

You may want to do a check for "